README.md

```diff
@@ -328,66 +349,165 @@ The complete list of transformations supported and their usage in ImageKit can b
+```
+
+## Constructing Video URLs
+The `ImageKitURLConstructor` can also be used to create a url that can be used for streaming videos with real-time transformations. `ImageKitURLConstructor` consists of functions that can be chained together to perform transformations.
+
+The initialization is same as that for image URLs by calling `ImageKit.getInstance().url(...)` with a set of parameters defined below.
+
+### Basic Examples
+```kotlin
+// Kotlin
+// https://ik.imagekit.io/your_imagekit_id/default-video.mp4?tr=h-400.00,w-400.00
+ImageKit.getInstance()
+    .url(
+        path = "default-video.mp4",
+        transformationPosition = TransformationPosition.QUERY
+    )
+    .height(400f)
+    .width(400f)
+    .create()
+```
+
+### Adaptive bitrate streaming
+To obtain the video URL with adaptive streaming, call `ImageKit.getInstance().url(...).setAdaptiveStreaming(...)` with a set of parameters defined below.
+
+| Parameter   | Type            | Description                                                                                                               |
+|:------------|:----------------|:--------------------------------------------------------------------------------------------------------------------------|
+| format      | StreamingFormat | Specifies the format for streaming video. Supported values for type are `StreamingFormat.HLS` and `StreamingFormat.DASH`. |
+| resolutions | List\<Int>      | Specifies the representations of the required video resolutions. E. g. 480, 720, 1080 etc.                                |
+
+Code example:
+```kotlin
+// Kotlin
+// https://ik.imagekit.io/your_imagekit_id/default-video.mp4/ik-master.m3u8?tr=sr-240_360_480_720_1080_1440_2160
+ImageKit.getInstance()
+    .url(
+        path = "default-video.mp4",
+        transformationPosition = TransformationPosition.QUERY
+    )
+    .setAdaptiveStreaming(
+        format = StreamingFormat.HLS,
+        resolutions = listOf(240, 360, 480, 720, 1080, 1440, 2160)
+    )
+    .create()
+```
+
+```java
+// Java
+// https://ik.imagekit.io/your_imagekit_id/default-video.mp4/ik-master.m3u8?tr=sr-240_360_480_720_1080_1440_2160
+ImageKit.Companion.getInstance()
+    .url(
+        "default-video.mp4",
+        TransformationPosition.QUERY
+    )
+    .setAdaptiveStreaming(StreamingFormat.HLS,
+        Arrays.asList(240, 360, 480, 720, 1080, 1440, 2160)
+    )
+    .create();
+```
+
+## File Upload
 The SDK provides a simple interface using the `ImageKit.getInstance().uploader().upload(...)` method to upload files to the ImageKit Media Library. It accepts all the parameters supported by the [ImageKit Upload API](https://docs.imagekit.io/api-reference/upload-file-api/client-side-file-upload#request-structure-multipart-form-data).
 
 Make sure that you have specified `authenticationEndpoint` during SDK initialization. The SDK makes an HTTP GET request to this endpoint and expects a JSON response with three fields i.e. `signature`, `token`, and `expire`.  
```

Comment:
Let's change this. In this new SDK:
* Use [secure client side upload - V2](https://docs.imagekit.io/api-reference/upload-file-api/secure-client-side-file-upload)
* Let user of the SDK generate token however they wish and make `token` parameter compulsory in upload API payload. Don't ask for `authenticationEndpoint` and internally hit that endpoint to get `tokent` like we do right now. I can explain more about this on the call and why we are changing this.

Reasons for changing this:
* V2 uses JWT based authentication which verifies the whole payload content.
* Our SDK call `authenticationEndpoint` without any custom header, so for users its still unsecured endpoint at their end. They want to handle authentication themself, so SDK should not get in the way.

Reply:
Alright, will change this after discussing this on the call

Reply:
Done